pkg/mimirtool/rules/compare.go

```diff
@@ -132,6 +134,14 @@ func CompareGroups(groupOne, groupTwo rwrulefmt.RuleGroup) error {
 		return errDiffRWConfigs
 	}
 
+	if ((groupOne.EvaluationDelay == nil) != (groupTwo.EvaluationDelay == nil)) || (groupOne.EvaluationDelay != nil && groupTwo.EvaluationDelay != nil && *groupOne.EvaluationDelay != *groupTwo.EvaluationDelay) {
```

Comment:
`groupOne.EvaluationDelay == nil` should be treated the same as `groupOne.EvaluationDelay != nil && *groupOne.EvaluationDelay == 0` as we do not store the zero value, so the tool could get into an infinite loop trying to sync the 0 value.

Reply:
Good point. I should have addressed it here: https://github.com/grafana/mimir/pull/8297/commits/e9baaa2bba4ef515b32651e617b8aee391c65216. It also allowed me to simplify the comparison.